### PR TITLE
Base AiiDAlab image on jupyter/base-notebook image and support AiiDA 2.0

### PR DESCRIPTION
**Pull request by yakutovicha**
- [x] https://github.com/aiidalab/aiidalab/pull/301 (https://github.com/aiidalab/aiidalab/milestone/2)
- [x] https://github.com/aiidalab/aiidalab-home/pull/102
- [x] Revise tests (the `wait-for-services` executable is no longer present)
- [x] ~Mount PostgreSQL and RabbitMQ data folders into home directory (to keep AiiDA data self-contained)~ ([not possible](https://github.com/aiidalab/aiidalab-docker-stack/pull/254#issuecomment-1219788707))
- [x] Install aiidalab package from conda-forge

_Edited by @csadorf_

### CODE REVIEW

**Comment by csadorf** [comment]
This PR is almost ready to go, we have the following outstanding issues:

- [ ] https://github.com/aiidalab/aiidalab/pull/301 (https://github.com/aiidalab/aiidalab/milestone/2)
- [ ] Revise tests (the `wait-for-services` executable is no longer present)

File: Dockerfile
Comment (csadorf) [inline Dockerfile:60]: - [ ] Switch to release version after release of https://github.com/aiidalab/aiidalab/pull/301

**Comment by csadorf** [comment]
Blocked by: https://github.com/conda-forge/staged-recipes/pull/19940

**Comment by csadorf** [comment]
> @csadorf I guess this is no longer blocked?

Kind of. We'll have to make a release of https://github.com/aiidalab/aiidalab and then we can switch to installing from conda. I'm also trying to merge https://github.com/aiidalab/aiidalab-docker-stack/pull/276, but I'm having some trouble getting GitHub actions to actually build the aiidalab image. Feel free to have a look, maybe you can spot the problem?

The refactoring into an actual hierarchical stack will enable me to implement the one-off migration image needed for the switch to AiiDA 2.0 and the slightly changed user environment which is the last task on https://github.com/aiidalab/aiidalab-docker-stack/issues/240.

**Comment by ltalirz** [comment]
> I'm also trying to merge #276, but I'm having some trouble getting GitHub actions to actually build the aiidalab image. Feel free to have a look, maybe you can spot the problem?

The Dockerfile is starting from an `aiidalab/aiida` image. I don't see this on https://hub.docker.com/u/aiidalab

**Comment by csadorf** [comment]
> The Dockerfile is starting from an `aiidalab/aiida` image. I don't see this on https://hub.docker.com/u/aiidalab

Thanks for checking! The image is built locally in the step prior ("Build aiida image"), however I think I finally identified the underlying problem: the buildx context is probably not using the docker driver (compare https://github.com/docker/buildx/issues/301). I'd suggest to move any further discussion to https://github.com/aiidalab/aiidalab-docker-stack/pull/276 .

**Review by csadorf — APPROVED** [review]

**Comment by csadorf** [comment]
@yakutovicha This is ready to go, do you want to have another look at it before I merge?

File: environment.yml
Comment (yakutovicha) [inline environment.yml:13]: This should probably be changed.

Comment (csadorf) [reply environment.yml:13]: Yes.

File: stack/base/requirements.txt
Comment (csadorf) [reply stack/base/requirements.txt:1]: This prevents the user from installing an aiida-core version < 2 with pip. However, maybe we should constrain it like this:
```suggestion
aiida-core>=2.0.0,<3
```

Comment (yakutovicha) [reply stack/base/requirements.txt:1]: sounds good 👍

**Comment by csadorf** [comment]
@yakutovicha I'll merge this, ok?

**Comment by yakutovicha** [comment]
> @yakutovicha I'll merge this, ok?

sure, got ahead.

**Comment by csadorf** [comment]
> [EDIT] please either add the ssh-agent as I mentioned above or make an issue to not forget about this. People are actually using it.

@yakutovicha Implemented in 323bb51, I'm not sure how to test it.

**Comment by csadorf** [comment]
@yakutovicha I'll merge this, ok?

**Comment by yakutovicha** [comment]
> @yakutovicha I'll merge this, ok?

yes, please go ahead

**Pull request merged**
